docs/release-info/release-notes.md

```diff
@@ -57,50 +57,765 @@ For tips about how to write a good release note, see [Release notes](https://git
+
+Added `SCALAR_IN_ARRAY` function for checking if a scalar expression appears in an array:
+
+`SCALAR_IN_ARRAY(expr, arr)`
+
+[#16306](https://github.com/apache/druid/pull/16306)
+
+#### Improved PARTITIONED BY
+
+If you use the MSQ task engine to run queries, you can now use the following strings in addition to the supported ISO 8601 periods:
+
+- `HOUR` - Same as `'PT1H'`
+- `DAY` - Same as `'P1D'`
+- `MONTH` - Same as `'P1M'`
+- `YEAR` - Same as `'P1Y'`
+- `ALL TIME`
+- `ALL` - Alias for `ALL TIME`
+
+[#15836](https://github.com/apache/druid/pull/15836/)
+
+#### Improved filter bundles
+
+Improved filter bundles as follows:
+
+* Renamed the parameter `selectionRowCount` on `makeFilterBundle` to `applyRowCount`, and redefined as an upper bound on rows remaining after short-circuiting (rather than number of rows selected so far).
+This definition works better for OR filters, which pass through the
+FALSE set rather than the TRUE set to the next subfilter.
+* `AndFilter` uses `min(applyRowCount, indexIntersectionSize)` rather than using `selectionRowCount` for the first subfilter and `indexIntersectionSize` for each filter thereafter. This improves accuracy when the incoming `applyRowCount` is smaller than the row count from the first few indexes.
+* `OrFilter` uses `min(applyRowCount, totalRowCount - indexUnionSize)` rather than `applyRowCount` for subfilters. This allows an OR filter to pass
+information about short-circuiting to its subfilters.
+
+[#16292](https://github.com/apache/druid/pull/16292)
+
+#### Improved catalog tables
+
+You can validate complex target column types against source input expressions during DML INSERT/REPLACE operations.
+
+[#16223](https://github.com/apache/druid/pull/16223)
+
+You can now define catalog tables without explicit segment granularities. 
+DML queries on such tables need to have the PARTITIONED BY clause specified.
+Alternatively, you can update the table to include a defined segment granularity for DML queries to be validated properly.
+
+[#16278](https://github.com/apache/druid/pull/16278)
+
+#### Double and null values in SQL type ARRAY
+
+You can now pass double and null values in SQL type ARRAY through dynamic parameters.
+
+For example:
+
+```json
+"parameters": [
+  {
+    "type": "ARRAY",
+    "value": [d1, d2, null]
+  }
+]
+```
+
+[#16274](https://github.com/apache/druid/pull/16274)
+
+#### `TypedInFilter` filter
+
+Added a new `TypedInFilter` filter to replace `InDimFilter`&mdash;to improve performance when matching numeric columns.
+
+[#16039](https://github.com/apache/druid/pull/16039)
+
+`TypedInFilter` can run in replace-with-default mode.
+
+[#16233](https://github.com/apache/druid/pull/16233)
+
+#### Improved partial index value matching for OR filter
+
+Partial index value matchers for the OR filter now use `PeekableIntIterator` instead of `IntIterator`.
+This change can significantly improve performance when Druid uses the value matchers alongside an index offset.
```

Comment:
I'm not sure how much end users may understand from this; wouldn't it enough that processing of index backed `OR` expressions was improved?

Reply:
Updated